profiles/management/commands/gen_realistic_users.py

```diff
@@ -0,0 +1,254 @@
+"""
+Generates a set of realistic users to help us test search functionality
+"""
+from datetime import datetime, timedelta
+from django.core.management import BaseCommand
+from django.contrib.auth.models import User
+from profiles.models import Employment, Education
+from courses.models import Program, Course, CourseRun
+from dashboard.models import CachedCertificate, CachedEnrollment
+from micromasters.utils import load_json_from_file
+
+
+USER_DATA_PATH = 'profiles/management/realistic_user_data.json'
+PROGRAM_DATA_PATH = 'profiles/management/realistic_program_data.json'
+FAKE_USER_USERNAME_PREFIX = 'fake.'
+FAKE_PROGRAM_DESC_PREFIX = '[FAKE] '
+CACHED_MODEL_LAST_REQUEST = datetime.now() + timedelta(days=365)
+
+
+# Util functions
+
+def first_matching_item(iterable, predicate):
+    """Returns the first item in an iterable that matches a given predicate"""
+    return next(x for x in iterable if predicate(x))
+
+
+def is_relation_field(model_obj, attr_name):
+    """
+    Checks if a given model attribute is a relation rather than a single value
+    """
+    return 'RelatedManager' in str(type(getattr(model_obj, attr_name)))
```

Comment:
this exists for the benefit of `deserialize_model_data`. in that method, i have a model object and some data (key-value pairs). all it's trying to do is set the value of a given attribute if that attribute is a scalar value (int/string/boolean/etc). if that `data` dict has a key that refers to a 'related' field - eg: `work_history` in a Profile object - i don't want to attempt to set it. this method works in the cases that i'm using it. that User.profile case is an interesting one. i'll have a look at that and see if i can make this function more 'correct'